Dockerfile

```diff
@@ -1,111 +1,233 @@
-RUN chmod +x /usr/bin/run_*
-
-# Replace imglob with a Python3 compatible version
-COPY nipype/external/fsl_imglob.py /usr/bin/fsl_imglob.py
-RUN rm -rf ${FSLDIR}/bin/imglob && \
-    chmod +x /usr/bin/fsl_imglob.py && \
-    ln -s /usr/bin/fsl_imglob.py ${FSLDIR}/bin/imglob
-
-# Installing dev requirements (packages that are not in pypi)
-WORKDIR /src/
-COPY requirements.txt requirements.txt
-RUN pip install -r requirements.txt && \
-    rm -rf ~/.cache/pip
-
-RUN git clone https://github.com/INCF/pybids.git && \
-    cd pybids && python setup.py develop
-
-# Installing nipype
-COPY . /src/nipype
-RUN cd /src/nipype && \
-    pip install -e .[all] && \
-    rm -rf ~/.cache/pip
-
-WORKDIR /work/
+ARG DEBIAN_FRONTEND=noninteractive
+
+#----------------------------------------------------------
+# Install common dependencies and create default entrypoint
+#----------------------------------------------------------
+ENV LANG="en_US.UTF-8" \
+    LC_ALL="C.UTF-8" \
+    ND_ENTRYPOINT="/neurodocker/startup.sh"
+RUN apt-get update -qq && apt-get install -yq --no-install-recommends  \
+    	apt-utils bzip2 ca-certificates curl locales unzip \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* \
+    && localedef --force --inputfile=en_US --charmap=UTF-8 C.UTF-8 \
+    && chmod 777 /opt && chmod a+s /opt \
+    && mkdir -p /neurodocker \
+    && if [ ! -f "$ND_ENTRYPOINT" ]; then \
+         echo '#!/usr/bin/env bash' >> $ND_ENTRYPOINT \
+         && echo 'set +x' >> $ND_ENTRYPOINT \
+         && echo 'if [ -z "$*" ]; then /usr/bin/env bash; else $*; fi' >> $ND_ENTRYPOINT; \
+       fi \
+    && chmod -R 777 /neurodocker && chmod a+s /neurodocker
+ENTRYPOINT ["/neurodocker/startup.sh"]
+
+LABEL maintainer="The nipype developers https://github.com/nipy/nipype"
+
+ENV MKL_NUM_THREADS="1" \
+    OMP_NUM_THREADS="1"
+
+# Create new user: neuro
+RUN useradd --no-user-group --create-home --shell /bin/bash neuro
+USER neuro
+
+#------------------
+# Install Miniconda
+#------------------
+ENV CONDA_DIR=/opt/conda \
+    PATH=/opt/conda/bin:$PATH
+RUN echo "Downloading Miniconda installer ..." \
+    && miniconda_installer=/tmp/miniconda.sh \
+    && curl -sSL -o $miniconda_installer https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+    && /bin/bash $miniconda_installer -b -p $CONDA_DIR \
+    && rm -f $miniconda_installer \
+    && conda config --system --prepend channels conda-forge \
+    && conda config --system --set auto_update_conda false \
+    && conda config --system --set show_channel_urls true \
+    && conda clean -tipsy && sync
+
+#-------------------------
+# Create conda environment
+#-------------------------
+RUN conda create -y -q --name neuro \
+    && sync && conda clean -tipsy && sync \
+    && sed -i '$isource activate neuro' $ND_ENTRYPOINT
 
+COPY ["docker/files/run_builddocs.sh", "docker/files/run_examples.sh", "docker/files/run_pytests.sh", "nipype/external/fsl_imglob.py", "/usr/bin/"]
```

Comment:
i don't feel to strongly about changing this, but maybe we can take advantage of `--chown` flag now

Reply:
yes we can do that. although this would require an update to neurodocker. could this be done in a future pr?

Reply:
yes